src/selectors.jl

```diff
@@ -104,8 +104,10 @@ Base.show(io::IO, ::MIME"text/plain", s::Near{T}) where {T} =
 findindex(sel::Near, range::AbstractArray) = argmin(map(x -> abs(x-sel.val), range))
 
 function findindex(sel::Near, range::AbstractRange)
-    iplus = searchsortedfirst(range, sel.val)
+    iplus = searchsortedfirst(range, sel.val; rev=step(range) < zero(step(range)))
     # "index of the first value in a greater than or equal to x"
+    iplus == 1 && return 1
```

Comment:
This function is a real mess, sorry! 

I think this will make it allow for offsets:
```suggestion
    iplus == firstindex(range) && return 1
```